Framework/DataHandling/test/LoadInstrumentTest.h

```diff
@@ -517,6 +601,49 @@ class LoadInstrumentTest : public CxxTest::TestSuite {
     TS_ASSERT(!instLoader.execute())
   }
 
+  void test_failure_if_Filename_not_set() {
+    LoadInstrument instLoader;
+    instLoader.initialize();
+    instLoader.setChild(true);
+    MatrixWorkspace_sptr ws =
+        DataObjects::create<Workspace2D>(1, HistogramData::Points(1));
+    instLoader.setProperty("Workspace", ws);
+    instLoader.setProperty("RewriteSpectraMap", OptionalBool(true));
+    TS_ASSERT_THROWS_EQUALS(
+        instLoader.execute(), Kernel::Exception::FileError & e,
+        std::string(e.what()),
+        "Either the InstrumentName or Filename property of LoadInstrument "
+        "must be specified to load an instrument in ");
+    TS_ASSERT(!instLoader.isExecuted());
+    TS_ASSERT_EQUALS(instLoader.getPropertyValue("Filename"), "");
+  }
+
+  void test_failure_if_Filename_not_found() {
+    LoadInstrument instLoader;
+    instLoader.initialize();
+    MatrixWorkspace_sptr ws =
+        DataObjects::create<Workspace2D>(1, HistogramData::Points(1));
+    instLoader.setProperty("Workspace", ws);
+    instLoader.setPropertyValue("Filename", "Nonsense");
+    instLoader.setProperty("RewriteSpectraMap", OptionalBool(true));
+    TS_ASSERT_THROWS_EQUALS(
```

Comment:
somehow, `ctest` is not picking up the error here. It reports that the test did not throw, even though I do have the message "FileDescriptor() - File 'Nonsense' does not exist" in my log.
Any ideas?

Reply:
Try adding `instLoader.setRethrows(true)` after initialize.

Reply:
:+1: